x-pack/plugins/ui_actions_enhanced/public/drilldowns/url_drilldown/url_template.ts

```diff
@@ -69,6 +70,63 @@ handlebars.registerHelper('date', (...args) => {
   return format ? momentDate.format(format) : momentDate.toISOString();
 });
 
+handlebars.registerHelper('formatNumber', (rawValue: unknown, pattern: string) => {
+  if (!pattern || typeof pattern !== 'string')
+    throw new Error(`[formatNumber]: pattern string is required`);
+  const value = Number(rawValue);
+  if (rawValue == null || Number.isNaN(value)) return rawValue;
+  return numeral(value).format(pattern);
+});
+
+/**
+ * Allows to match regex patterns and extract capturing groups.
+ * Result is array of arrays.
+ *
+ * @example
+ *
+ * Have a string: "Label:Feature:Something"
+ * and want to extract: "Feature:Something"
+ *
+ * expression: `{{match value "Label:(.*)"}}`,
+ * returns: [["Label:Feature:Something", "Feature:Something"]]
+ */
+handlebars.registerHelper('match', (rawValue: unknown, regexpString: string) => {
+  if (!regexpString || typeof regexpString !== 'string')
+    throw new Error(`[match]: regexp string is required`);
+  const regexp = new RegExp(regexpString, 'g');
+  const valueString = String(rawValue);
+  return Array.from(valueString.matchAll(regexp));
+});
+
+function toString(value: unknown): string {
+  return String(value);
+}
+handlebars.registerHelper('lowercase', (rawValue: unknown) => toString(rawValue).toLowerCase());
```

Comment:
Curious, does wrapping `String()` into `toString()` change something here?

```suggestion
handlebars.registerHelper('lowercase', (rawValue: unknown) => String(rawValue).toLowerCase());
```